echopype/utils/coding.py

```diff
@@ -35,6 +35,23 @@
 }
 
 
+EXPECTED_VAR_DTYPE = {"channel": np.str_, "beam": np.str_}  # channel name  # beam name
+
+
+def sanitize_dtypes(ds: xr.Dataset) -> xr.Dataset:
+    """
+    Validates and fixes data type for expected variables
+    """
+
+    if isinstance(ds, xr.Dataset):
+        for name, var in ds.variables.items():
+            if name in EXPECTED_VAR_DTYPE:
+                expected_dtype = EXPECTED_VAR_DTYPE[name]
+                if not np.issubdtype(var.dtype, expected_dtype):
+                    ds[name] = var.astype(expected_dtype)
+    return ds
```

Comment:
This new `sanitize_dtypes` method will ensure that the data type of each variable is the expected one. Currently, the `channel` and `beam` data variable is `object` dtype and when trying to output to zarr it fails in `echopype.utils.coding.get_zarr_compression` since this dtype is not listed. So let's cast it to ensure that it is the correct data type.

Reply:
This is great -- and I didn't know there is `np.issubdtype`!